src/rand_impls.rs

```diff
@@ -269,9 +269,38 @@ mod tests {
 
     #[test]
     fn floating_point_edge_cases() {
-        // the test for exact equality is correct here.
-        assert!(ConstantRng(0xffff_ffff).gen::<f32>() != 1.0);
-        assert!(ConstantRng(0xffff_ffff_ffff_ffff).gen::<f64>() != 1.0);
+        const EPSILON32: f32 = 1.0 / (1u32 << 23) as f32;
+        const EPSILON64: f64 = 1.0 / (1u64 << 52) as f64;
```

Comment:
Why not use `core::f32::EPSILON`?

ε is defined as the difference between 1.0 and the next largest representable number; since all numbers output (excepting `Closed01`) are strictly less than 1.0 the exponent will be 1 less than for 1.0, hence the largest representable number less than 1 should be `1 - ε / 2`.

Reply:
I first used `std::f32::EPSILON` but that failed the continuous integration tests and I forgot about `core`, so I'll change this.